src/api/middleware.js

```diff
@@ -106,7 +106,11 @@ export function allow(auth: IAuth) {
     return function(req: $RequestExtend, res: $ResponseExtend, next: $NextFunctionVer) {
       req.pause();
       const packageName = req.params.scope ? `@${req.params.scope}/${req.params.package}` : req.params.package;
-      const packageVersion = req.params.filename ? getVersionFromTarball(req.params.filename) : undefined;
+      const packageVersion = req.params.filename
+        ? getVersionFromTarball(req.params.filename)
+        : req.body && req.body.versions
```

Comment:
```
req.body.versions
```
From where this information is coming from? 🤔

Reply:
It comes from npm at `npm publish`

/api/middleware.js:
```
function allow(auth) {
  return function (action) {
    return function (req, res, next) {
      req.pause();
      console.log(`req.params`, req.params);
      console.log(`req.body`, req.body);
...
```

Log:
```
req.params { package: '@latoken-component/plugin1',
  _rev: undefined,
  revision: undefined }
req.body { _id: '@latoken-component/plugin1',
  name: '@latoken-component/plugin1',
  description: '',
  'dist-tags': { latest: '1.0.3-beta.3' },
  versions:
   { '1.0.3-beta.3':
      { name: '@latoken-component/plugin1',
        version: '1.0.3-beta.3',
        description: '',
        main: 'index.js',
        scripts: [Object],
        author: '',
        license: 'ISC',
        readme: 'ERROR: No README data found!',
        _id: '@latoken-component/plugin1@1.0.3-beta.3',
        _npmVersion: '6.1.0',
        _nodeVersion: '10.4.1',
        _npmUser: {},
        dist: [Object] } },
  readme: 'ERROR: No README data found!',
  _attachments:
   { '@latoken-component/plugin1-1.0.3-beta.3.tgz':
      { content_type: 'application/octet-stream',
        data:
         'H4sIAAAAA....',
        length: 261 } } }

```